himbaechel/uarch/gatemate/gatemate.h

```diff
@@ -83,6 +85,8 @@ struct GateMateImpl : HimbaechelAPI
     pool<IdString> multiplier_a_passthru_uppers;
     pool<IdString> multiplier_zero_drivers;
     std::vector<CellInfo *> multipliers;
+    std::set<int32_t> used_cpes;
```

Comment:
any specific reason for `std::set` rather than `pool`? a flat vector of bool (one index per tile) might even be the most efficient